tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -45,17 +46,50 @@
 import roslaunch.parent
 
 from rosmaster.master import Master
+from rospy import logwarn
+
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, reuse_master=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if not reuse_master:
+            run_id = roslaunch.core.generate_run_id()
+        else:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except Exception as e:
+                logwarn("Could not connect to existing ROS master; will create a new master: " +
+                        str(e))
+            except:
+                # oh boy; we got something that wasn't an exception...
+                # ignore it, the same way that roslaunch does in this situation
+                logwarn("Could not connect to existing ROS master; will create a new master")
```

Comment:
The user explicitly asked for reusing an existing master. If that is not possible it should not be silently done imo.